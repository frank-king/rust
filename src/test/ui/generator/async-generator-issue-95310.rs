#![crate_type="lib"]
#![feature(generators)]
#![feature(generic_associated_types)]
#![feature(type_alias_impl_trait)]

// edition:2021

trait Service {
    type Future<'f>: std::future::Future + 'f
    where
        Self: 'f;

    fn spawn<'f>(&'f mut self) -> Self::Future<'f>;
}

struct Blah;

impl Service for Blah {
    type Future<'f> = impl std::future::Future<Output = &'static i32> + 'f;

    fn spawn<'f>(&'f mut self) -> Self::Future<'f> {
        //~^ ERROR: type mismatch resolving
        async move {
            //~^ ERROR: type mismatch resolving 
            yield &1; //~ ERROR: `async` generators are not yet supported
        }
    }
}