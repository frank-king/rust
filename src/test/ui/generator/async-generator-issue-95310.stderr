error[E0727]: `async` generators are not yet supported
  --> $DIR/async-generator-issue-95310.rs:25:13
   |
LL |             yield &1;
   |             ^^^^^^^^

error[E0271]: type mismatch resolving `<[static generator@$DIR/async-generator-issue-95310.rs:23:20: 26:10] as Generator<ResumeTy>>::Yield == ()`
  --> $DIR/async-generator-issue-95310.rs:23:20
   |
LL |           async move {
   |  ____________________^
LL | |
LL | |             yield &1;
LL | |         }
   | |_________^ expected `()`, found `&{integer}`
   |
note: required by a bound in `from_generator`
  --> $SRC_DIR/core/src/future/mod.rs:LL:COL
   |
LL |     T: Generator<ResumeTy, Yield = ()>,
   |                            ^^^^^^^^^^ required by this bound in `from_generator`

error[E0271]: type mismatch resolving `<impl Future<Output = [async output]> as Future>::Output == &'static i32`
  --> $DIR/async-generator-issue-95310.rs:21:35
   |
LL |     fn spawn<'f>(&'f mut self) -> Self::Future<'f> {
   |                                   ^^^^^^^^^^^^^^^^ expected `&i32`, found `()`

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0271, E0727.
For more information about an error, try `rustc --explain E0271`.
